### PR TITLE
Update to latest github-api 1.99

### PR DESCRIPTION
**Pull request by bitwiseman**
Also update to latest Jackson2 API and Okio.

Okio 2 is binary compatible with 1.x.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -31,12 +33,12 @@
...
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>jackson2-api</artifactId>
-      <version>2.9.7.1</version>
+      <version>2.10.0</version>
     </dependency>
     <dependency>
       <groupId>org.kohsuke</groupId>
       <artifactId>github-api</artifactId>
-      <version>1.95</version>
+      <version>1.99</version>
```
Comment (jglick) [inline pom.xml]: Tip: `requireProperty` as in https://github.com/jenkinsci/incrementals-tools#superseding-maven-releases

File: pom.xml [hunk 2]
```diff
@@ -56,6 +58,11 @@
         </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>com.squareup.okio</groupId>
+      <artifactId>okio</artifactId>
+      <version>2.4.0</version>
```
Comment (jglick) [inline pom.xml]: Why is this here? Is it an unexpressed dependency of some existing direct dependency?

File: src/test/java/jenkins/plugins/github/api/SmokeTest.java [hunk 3]
```diff
@@ -19,6 +21,10 @@
 import static org.junit.Assert.assertThat;
 
 public class SmokeTest {
+
+    @Rule
+    public JenkinsRule jenkins = new JenkinsRule();
```
Comment (jglick) [inline src/test/java/jenkins/plugins/github/api/SmokeTest.java]: Why?

Comment (bitwiseman) [reply src/test/java/jenkins/plugins/github/api/SmokeTest.java]: Because tests were failing without it?

Comment (KostyaSha) [reply src/test/java/jenkins/plugins/github/api/SmokeTest.java]: Existing tests checking only library itself, where Jenkins instance is neeed?

Comment (jglick) [reply src/test/java/jenkins/plugins/github/api/SmokeTest.java]: Yes I get that, but my question is _why_ tests would have been failing. Does this upgrade somehow introduce calls to `Jenkins.get` into tests which otherwise do not mention it? Or were they failing before this patch?

So now I see errors such as

```
java.lang.IllegalStateException: Jenkins.instance is missing. Read the documentation of Jenkins.getInstanceOrNull to see what you are doing wrong.
	at jenkins.model.Jenkins.get(Jenkins.java:758)
	at jenkins.telemetry.impl.StaplerDispatches$StaplerTrace.record(StaplerDispatches.java:105)
	at org.kohsuke.stapler.EvaluationTrace$ApplicationTracer.trace(EvaluationTrace.java:77)
	at org.kohsuke.stapler.Dispatcher.anonymizedTraceEval(Dispatcher.java:76)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:685)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:870)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:668)
	at org.kohsuke.stapler.Stapler.service(Stapler.java:238)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at …
```

which means that https://github.com/jenkinsci/github-api-plugin/blob/356484ad3f26f20d6a8f0c47a2196db188bb0e89/src/test/java/jenkins/plugins/github/api/mock/MockGitHub.java#L61 plus https://github.com/jenkinsci/jenkins/pull/3688 is causing problems and suggests that you need to update to 2.150.x to pick up https://github.com/jenkinsci/jenkins/pull/3712.

File: pom.xml (hunk 1 above)
Comment (bitwiseman) [reply pom.xml]: I'm sorry, what are you asking to be done here?

Comment (jglick) [reply pom.xml]: Just a suggestion, not a request: that you can make sure `version` and `revision` do not drift out of synch by accident.

File: src/test/java/jenkins/plugins/github/api/SmokeTest.java (hunk 3 above)
Comment (bitwiseman) [reply src/test/java/jenkins/plugins/github/api/SmokeTest.java]: Awesome!  Thanks @jglick !

File: pom.xml [hunk 4]
```diff
@@ -23,11 +23,45 @@
...
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <artifactId>maven-release-plugin</artifactId>
+        <configuration>
+          <!-- we do not want to auto-increment when releasing -->
+          <!-- use instructions at https://github.com/jenkinsci/incrementals-tools#superseding-maven-releases to release -->
+          <preparationGoals>not-set-up-for-MRP</preparationGoals>
```
Comment (jglick) [inline pom.xml]: No delete this stuff or you will not be able to release! I was only pointing you to the `maven-enforcer-plugin` configuration.

File: src/test/java/jenkins/plugins/github/api/SmokeTest.java [hunk 5]
```diff
@@ -8,7 +8,9 @@
 import jenkins.plugins.github.api.mock.MockOrganization;
 import jenkins.plugins.github.api.mock.MockUser;
 import org.apache.commons.io.IOUtils;
+import org.junit.Rule;
```
Comment (jglick) [inline src/test/java/jenkins/plugins/github/api/SmokeTest.java]:
```suggestion
```

File: src/test/java/jenkins/plugins/github/api/SmokeTest.java [hunk 6]
```diff
@@ -8,7 +8,9 @@
...
 import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
```
Comment (jglick) [inline src/test/java/jenkins/plugins/github/api/SmokeTest.java]:
```suggestion
```

File: src/test/java/jenkins/plugins/github/api/SmokeTest.java [hunk 7]
```diff
@@ -19,6 +21,7 @@
 import static org.junit.Assert.assertThat;
 
 public class SmokeTest {
+
```
Comment (jglick) [inline src/test/java/jenkins/plugins/github/api/SmokeTest.java]:
```suggestion
```

File: pom.xml [hunk 8]
```diff
@@ -16,27 +16,63 @@
...
+        </configuration>
+      </plugin>
```
Comment (jglick) [inline pom.xml]: (this block, should have used a multiline comment to begin with)

**Comment by oleg-nenashev** [comment]
@bitwiseman @KostyaSha FYI GitHub API 1.99+ requires commons codec 1.13 due to https://github.com/github-api/github-api/pull/551 . It is above the version used by the Jenkins core, but you do not get an error during the exclusion: https://github.com/jenkinsci/github-api-plugin/blob/master/pom.xml#L77-L80 . Same applies to other exclusions. I am putting the new GitHub API Plugin release on hold, because there is a non-zero chance of nuking the GitHub plugin ecosystem by this release. I suggest reviewing it after DW-JW

**Comment by oleg-nenashev** [comment]
This pull request cannot be reverted automatically. I suggest doing it manually so that we do not keep the  potentially explosive version in the master branch. WDYT @bitwiseman ?

Also CC @batmat
